rclc/include/rclc/executor.h

```diff
@@ -23,6 +23,7 @@ extern "C"
 
 #include <stdio.h>
 #include <stdarg.h>
+#include <pthread.h>
```

Comment:
We cannot include this in the micro-ROS default user library since we do not have this header available on all platforms.

Can we make this conditional with a `#ifdef`?